### PR TITLE
avoid multiple initialization of rand

### PR DESCRIPTION
**Pull request by shogo82148**
*Issue #, if available:*

It looks that `math/rand` is initialized two times.

- https://github.com/aws/aws-xray-sdk-go/blob/dc1201085c0b1e64259ef8f7cb48954691212db9/utils/rand.go#L17
- https://github.com/aws/aws-xray-sdk-go/blob/dc1201085c0b1e64259ef8f7cb48954691212db9/utils/timer.go#L17

`math/rand.Seed` changes global state of users' program, so any SDK should not call this.

*Description of changes:*

I implemented a random value generator only used by the X-Ray SDK.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by bhautikpip** [comment]
Hi @shogo82148,

Thanks for opening up this PR. changes looks good! I might be able to merge it once we resolve SQL test case failure issue.

File: utils/rand.go [hunk 1]
```diff
@@ -9,12 +9,72 @@
...
+func newSeed() int64 {
+	var seed int64
+	if err := binary.Read(crand.Reader, binary.BigEndian, &seed); err != nil {
+		// fallback to timestamp
+		seed = time.Now().UnixNano()
+	}
+	return seed
+}
+
+func newGlobalRand() *rand.Rand {
```
Comment (bhautikpip) [inline utils/rand.go]: Can we add tests for this function which covers all the combination ?

**Comment by bhautikpip** [comment]
Hi @shogo82148 ,

would you mind resolving the conflict and adding few tests around the function ?

**Comment by shogo82148** [comment]
resolving the conflict is done!

**Pull request merged**